### DISCUSSION TITLE
Maintain a consistent run datetime across ops

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8500] [Q&A]
What's the best way to maintain a consistent run datetime across ops? I have ops writing to a hive-style S3 path and want to make sure they all agree on a single date that the run was started at. Easiest would be putting the date in the job config at job creation time, but I'm wondering if there's a more native way

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1644144744821389?thread_ts=1644144744.821389&cid=C01U954MEER).*

**Comment by yuhan** [comment]
There are several ways to accomplish this.
### Option 1
If all you care about is the job creation time, you can get that info from the optional `context` argument within an op:
```python
@op
def my_op(context):
    context.instance.get_run_stats(context.run_id)
```
If you're also looking for ways to get this info elsewhere, we have a similar Q&A here: https://github.com/dagster-io/dagster/discussions/8417

### Option 2
Most jobs that care about the execution time might be using partitions, which give you a place to use it to configure your ops with the datetime you provide, like this example: https://docs.dagster.io/concepts/partitions-schedules-sensors/partitions#date-partitioned-job
Once you have the datetime provided through run config, you can access it inside an op using `context.op_config`.


### Option 3
If you can execute the run in process (as opposed to multi process), you can pass a date as a resource and it should be constant when being accessed from different ops. This won't work with multiprocess which is default execution option, because when in multiple processes, we will initialize a resource separately in every op process that requires it.
```python
from dagster import repository, op, job, resource, ResourceDefinition
from datetime import datetime
import time


@op(required_resource_keys={"constant_value"})
def my_op1(context):
    time.sleep(2)
    context.log.info(context.resources.constant_value)


@op(required_resource_keys={"constant_value"})
def my_op2(context, foo):
    time.sleep(2)
    context.log.info(context.resources.constant_value)


@op(required_resource_keys={"constant_value"})
def my_op3(context, foo):
    time.sleep(2)
    context.log.info(context.resources.constant_value)


@job(
    resource_defs={
        "constant_value": ResourceDefinition.hardcoded_resource(
            datetime.now().strftime("%Y-%m-%d %H:%M:%S%z")
        )
    },
    config={"execution": {"config": {"in_process": {}}}},
)
def my_job():
    my_op3(my_op2(my_op1()))
```
<img width="1392" alt="Screen Shot 2022-06-20 at 3 39 59 PM" src="https://user-images.githubusercontent.com/4531914/174685452-05594d74-bfaf-4375-a5d8-4fa72ab42fbf.png">
